src/session/get-session.ts

```diff
@@ -4,7 +4,113 @@ import { NextRequest, NextResponse } from 'next/server';
 import { SessionCache, Session, get } from '../session';
 
 /**
- * Get the user's session from the request.
+ * Get the user's session from the server.
+ *
+ * **In the App Router:**
+ *
+ * In a route handler:
+ *
+ * ```js
+ * // app/my-api/route.js
+ * import { getSession } from '@auth0/nextjs-auth0';
+ *
+ * export async function GET() {
+ *   const { user } = await getSession();
+ *   return NextResponse.json({ foo: 'bar' });
+ * }
+ *
+ * // Or, it's slightly more efficient to use the `req`, `res` args if you're
+ * // using with another part of the SDK like `withApiAuthRequired` or `getAccessToken`
+ * import { getSession, withApiAuthRequired } from '@auth0/nextjs-auth0';
+ *
+ * const GET = withApiAuthRequired(async function GET(req) {
+ *   const res = new NextResponse();
+ *   const { user } = await getSession(req, res);
+ *   return NextResponse.json({ foo: 'bar' }, res);
+ * })
+ *
+ * export { GET }
+ * ```
+ *
+ * In a page or React Server Component:
+ *
+ * ```js
+ * // app/my-page/page.js
+ * import { getSession } from '@auth0/nextjs-auth0';
+ *
+ * export default async function MyPage({ params, searchParams }) {
+ *   const { user } = await getSession();
+ *   return <h1>My Page</h1>
+ * }
+ * ```
+ *
+ * **Note:** You can't write to the cookie in a React Server Component, so updates
+ * to the session like setting the expiry of the rolling session won't be persisted.
+ * For this we recommend touching the session in the middleware using `getSession` or
```

Comment:
```suggestion
 * For this, we recommend touching the session in the middleware using `getSession` or
```

Reply:
Yep - there are other alternatives (the list get's quite long) - I only added the Server Action to `updateSession` because it suits the use case (it doesn't for `getSession`)

Reply:
Does `touchSession` fit the use case here?

Reply:
[9081ca7](https://github.com/auth0/nextjs-auth0/pull/1376/commits/9081ca7a6c8a442b1a8fd854378ebafbf043ba6e)